File: src/tests/file-c/getcwd.c
/*
 * Copyright(c) The Maintainers of Nanvix.
 * Licensed under the MIT License.
 */

//==================================================================================================
// Configuration
//==================================================================================================

/* Must come first. */
#define _POSIX_C_SOURCE 200809 // strnlen()

//==================================================================================================
// Imports
//==================================================================================================

#include <assert.h>
#include <limits.h>
#include <stdio.h>
#include <string.h>
#include <unistd.h>

//==================================================================================================
// Imports
//==================================================================================================

// Tests whether we can get the current working directory.
void test_getcwd(void)
{
    fprintf(stderr, "testing getcwd() ... ");

    char pathbuf[PATH_MAX] = {0};

    // Get current working directory and assert result.
    char *cwd = getcwd(pathbuf, sizeof(pathbuf));
    assert(cwd != NULL);
    assert(cwd == pathbuf);
    size_t len = strnlen(cwd, PATH_MAX);
    assert((len > 0) && (len < PATH_MAX));

    fprintf(stderr, "passed\n");
}

File: src/tests/file-rust/src/unsafe/fs/getcwd.rs
// Copyright(c) The Maintainers of Nanvix.
// Licensed under the MIT License.

//==================================================================================================
// Imports
//==================================================================================================

use ::syscall::unistd;

//==================================================================================================
// Standalone Functions
//==================================================================================================

/// Tests whether we can get the current working directory using `getcwd()`.
pub fn test() {
    if let Err(error) = unistd::getcwd() {
        panic!("{error:?}");
    };
}
